File: 115/solution.py
from random import random
from math import *


def cal(n, m):
    num = [0 for i in range(1000)]
    num1 = [0 for i in range(1000)]
    num[0] = 1
    for i in range(1, n + 1):
        num[i] = num[i - 1] + num1[i - 1]
        if i >= m:
            for j in range(i - m, -1, -1):
                num1[i] += num[j]
    return num[n] + num1[n]


if __name__ == "__main__":
    for i in range(10, 1000):
        v = cal(i, 50)
        print(i, v)
        if v >= 1000000:
            break

File: 115/solution.go
package main

import "fmt"

var num, num1 = [1000]int64{}, [1000]int64{}

func cal(n, m int64) int64 {
	for i := 0; i < 1000; i++ {
		num[i], num1[i] = 0, 0
	}
	num[0] = 1
	for i := int64(1); i <= n; i++ {
		num[i] = num[i-1] + num1[i-1]
		if i >= m {
			for j := i - m; j >= 0; j-- {
				num1[i] += num[j]
			}
		}
	}
	return num[n] + num1[n]
}

func main() {
	for i := int64(10); ; i++ {
		value := cal(i, 50)
		fmt.Printf("%d %d\n", i, value)
		if value >= 1000000 {
			break
		}
	}
}
